### PR TITLE
handling errors in various files

### PR DESCRIPTION
**Pull request by erikvegax**
#9 

lmk if you guys see anything that needs fixing or changing

### CODE REVIEW

File: account/reward.go [hunk 1]
```diff
@@ -14,7 +14,13 @@ func AddCoins(rollno string, coins int) error {
 		return err
 	}
 
-	if !UserExists(rollno) {
+	userExists, err := UserExists(rollno)
+
+	if err != nil {
+		return err
```
Comment (1-Harshit) [inline account/reward.go]: Is this `return err` consistent with our `errors.NewHTTPError` ?

**Review by 1-Harshit — COMMENTED** [review]
error messages can be improved and make sure handlers return `errors.NewHTTPError`

File: account/reward.go (hunk 1 above)
Comment (bhuvansingla) [reply account/reward.go]: Yes, return an internal server error.

File: auth/login.go [hunk 2]
```diff
@@ -12,10 +12,24 @@ func Login(rollno string, password string) (ok bool, err error) {
 	if err = account.ValidateRollNo(rollno); err != nil {
 		return false, err
 	}
-	if !account.UserExists(rollno) {
-		return false, errors.NewHTTPError(nil, http.StatusBadRequest, "account does not exist")
+
+	userExists, err := account.UserExists(rollno)
+
+	if err != nil {
+		return false, errors.NewHTTPError(nil, http.StatusBadRequest, "error loggin in")
```
Comment (bhuvansingla) [inline auth/login.go]: This should be an internal server error, with the default message for internal server error.

Comment (bhuvansingla) [reply auth/login.go]: Here is an example: https://github.com/bhuvansingla/iitk-coin/blob/68efda6cef6e04ae8544d3d438bb07a5a7b92205/handlers/balance.go#L47

**Comment by erikvegax** [comment]
hey guys! how should i know when to return an internal server error vs. a regular error?

**Comment by erikvegax** [comment]
went through a used internal server error in the files that i was working on. lmk if you guys want me to fix anything else

**Comment by erikvegax** [comment]
also, i saw some errors where the string was capitalized. i think it's convention to have the sentence all lowercase as suggested by the [golang style guide](https://github.com/golang/go/wiki/CodeReviewComments#error-strings).

**Comment by 1-Harshit** [comment]
> hey guys! how should i know when to return an internal server error vs. a regular error?  

@erikvega when it is error from our side, computational or accessing database (not returning anything against query is another thing but failing while executing the query is an internal server error) etc...




> also, i saw some errors where the string was capitalized. i think it's convention to have the sentence all lowercase as suggested by the [golang style guide](https://github.com/golang/go/wiki/CodeReviewComments#error-strings).

That's true, if u see them pls fix them or highlight it

File: account/user.go [hunk 3]
```diff
@@ -79,7 +88,7 @@ func ValidatePassword(password string) error {
 }
 
 func IsAdmin(rollno string) bool {
```
Comment (bhuvansingla) [inline account/user.go]: @erikvega can you also modify this method to check for errors that might get passed on from `GetAccountRoleByRollno` on L91?

**Review by bhuvansingla — CHANGES_REQUESTED** [review]

**Comment by erikvegax** [comment]
i went ahead and added the changes that @bhuvansingla requested. i'll let one of you guys handle the conflict because you will know what's best. any other changes?

**Review by bhuvansingla — APPROVED** [review]
LGTM.

**Pull request merged**